### DISCUSSION TITLE
Change Neo-Tree mappings and key bindings to behave more like nnn

### DISCUSSION THREAD

**Discussion by rafo** [LazyVim/LazyVim#1492] [Q&A] [answered]
I use [nnn](https://github.com/jarun/nnn) as my go-to file explorer in the terminal and want to change some [keyboard settings in neo-tree](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/defaults.lua) to avoid memorizing an additional set of key bindings. The [nnn key-bindings](https://github.com/jarun/nnn/wiki/Usage#keyboard-mouse) are very minimal and quite intuitive (imho). Yes, I tried the nnn.nvim plugin. This would be the perfect solution in theory, but it has its limitations.

So I tryed to remap
"." to "toggle_hidden"
"l" to "open"

with many variants with no luck. Right now setting the root with the default "." doesnt work anymore. My .config/nvim/lua/plugins/neo-tree.lua:

```
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    opts = {
      sources = { "filesystem", "buffers", "git_status", "document_symbols" },
      open_files_do_not_replace_types = { "terminal", "Trouble", "qf", "Outline" },
      filesystem = {
        bind_to_cwd = false,
        follow_current_file = { enabled = true },
        use_libuv_file_watcher = true,
        filtered_items = {
          hide_dotfiles = false,
        },
        window = {
          mappings = {
            ["."] = "toggle_hidden",
            ["l"] = "open",
            ["L"] = "open_nofocus",
            ["P"] = "focus_preview",
          },
        },
      },
      window = {
        mappings = {
          ["."] = "toggle_hidden", -- For testing This dont belong here
          ["<space>"] = "none",
        },
      },
      default_component_configs = {
        indent = {
          with_expanders = true, -- if nil and file nesting is enabled, will enable expanders
          expander_collapsed = "",
          expander_expanded = "",
          expander_highlight = "NeoTreeExpander",
        },
      },
    },
    --
    -- Is this really needed? I think it is not but https://github.com/folke/lazy.nvim#%EF%B8%8F-lazy-key-mappings
    config = function(_, opts)
      require("neo-tree").setup(opts)
    end,
  },
}
```

**Comment by dpetka2001** [comment, marked as answer]
```lua
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    opts = {
      filesystem = {
        window = {
          mappings = {
            ["."] = "toggle_hidden",
            ["H"] = "set_root",
          },
        },
      },
      window = {
        mappings = {
          ["l"] = "open",
          ["h"] = "close_node",
        },
      },
    },
  },
}
```
check this out if it works according to your preferences.

**Reply by rafo** [reply]
Nope.

"." does nothing
"l" does nothing
"h" does nothing
"H" does "toggle_hidden"

One thing I noticed: with "h" I can go to the left (like in vim) in the tree of neo-tree if my cursor is not already all the way to the left.

**Reply by rafo** [reply]
It works!

I disabled "luukvbaal/nnn.nvim", an did a _restart_. Is a restart necessary? I thought syncing lazy is enough?

**Reply by dpetka2001** [reply]
No, a restart is necessary, because Lua modules which are `require`d are cached. `lazy.nvim` provides an option to reload a module but I'm not aware if it provides an option to completely unload a module.
